src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -2337,5 +2341,44 @@ private IInstanceReferenceOperation CreateCollectionValuePlaceholderOperation(Bo
             bool isImplicit = placeholder.WasCompilerGenerated;
             return new InstanceReferenceOperation(referenceKind, _semanticModel, syntax, type, isImplicit);
         }
+
+        private ImmutableArray<IArgumentOperation> CreateDisposeArguments(MethodSymbol? disposeMethod, TypeSymbol? receiverType, SyntaxNode receiverSyntax, Binder binder)
+        {
+            if (disposeMethod is null || receiverType is null || disposeMethod.ParameterCount == 0)
```

Comment:
>receiverType is null [](start = 41, length = 20)

The fact that we can come here without the type feels unexpected. Shouldn't we always know the type?